src/server/game_server.rs

```diff
@@ -126,24 +144,49 @@ impl GameServer {
         }
     }
     
-    pub fn on_new_attack(&mut self, attack: EntityAttack) {
+    pub fn on_new_attack(&mut self, attack: EntityAttack, player_id: usize) {
         let victim = attack.victim_id();
-        // Communicate the attack to the player !
+        match attack.entity_kind() {
+            EntityKind::Player => {
+                // Communicate the attack to the player !
+                for i in 0..self.state.n_players_connected() {
+                    if i as u8 == victim {
+                        self.server_updates_buffer.get_mut(&i).unwrap().push(Attack(attack));
+                        return;
+                    }
+                }
+            }
+            _ => {
+                self.entity_server.remove_monster(victim as usize);
+                for i in 0..self.state.n_players_connected() {
+                    if i != player_id {
+                        self.server_updates_buffer.get_mut(&i).unwrap().push(RemoveEntity(victim as u32));
+                    }
+                }
+            }
```

Comment:
This is wrong, `n_plyers_connected` gives you the number of connected players, not their id !!!

Reply:
You can use the iterator function `connected`